reana_job_controller/job_monitor.py

```diff
@@ -122,28 +117,41 @@ def update_job_status(self, reana_job_id, status):
         """
         self.job_db[reana_job_id]["status"] = status
 
-    def should_process_job(self, job_pod):
+    def should_process_job(self, job_pod) -> bool:
         """Decide whether the job should be processed or not.
 
-        For a job to be processed it has to:
-        - Be a job created by this instance, that's why we check the in
-          memory DB.
-        - It has to be an active job, not having been deleted already.
-        - It has to be terminated, otherwise no status or logs can be
-          retrieved.
-
         :param job_pod: Compute backend job object (Kubernetes V1Pod
             https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Pod.md)
-
-        :return: Boolean representing whether the job should be processed or
-            not.
-        :rtype: bool
         """
-        remaining_jobs = self._get_remaining_jobs()
+        remaining_jobs = self._get_remaining_jobs(
+            statuses_to_skip=[JobStatus.finished.name, JobStatus.failed.name]
+        )
         backend_job_id = self.get_backend_job_id(job_pod)
-        is_job_in_memory_db = self.job_db.get(remaining_jobs.get(backend_job_id))
         is_job_in_remaining_jobs = backend_job_id in remaining_jobs
-        return is_job_in_memory_db or is_job_in_remaining_jobs
+
+        job_status = self.get_job_status(job_pod)
+        is_job_completed = job_status in [
+            JobStatus.finished.name,
+            JobStatus.failed.name,
+        ]
+
+        return (
+            is_job_in_remaining_jobs
+            and is_job_completed
+            and self._all_job_containers_not_running(job_pod)
```

Comment:
k8s sends multiple events with `phase=Failed` when a job is terminated due to timeout (maybe, in other situations too). In the first event, a container is still running but logs are not collected. In the second event, a container is terminated and logs are collected. That is why I need to add a check that all containers inside are not running to allow to process a job.